Source/AwsCommonRuntimeKit/auth/signing/SigV4HttpRequestSigner.swift

```diff
@@ -2,14 +2,123 @@
 //  SPDX-License-Identifier: Apache-2.0.
 
 import AwsCAuth
+import Foundation
 
-public class SigV4HttpRequestSigner {
-    public var allocator: Allocator
+//TODO: update file name
+public class HttpRequestSigner {
```

Comment:
Don't combine the signing-config class with the signer class. This is one of those things where the config is often passed around as its own thing, to be used later.

That said, `SigV4HttpRequestSigner` was a bad name for this class. We eventually want to support...:
- different signing algorithms (currently SigV4 and SigV4a, but there have been other algorithms in the past (SigV2, SigV3) and surely SigV5 will come some day.
- signing different types of objects (HTTP requests, chunks within an HTTP request, etc)

aws-crt-python has 1 [AwsSigningConfig](https://github.com/awslabs/aws-crt-python/blob/191b52d307793ac105d7389b0bce646917857ed1/awscrt/auth.py#L458) class, and currently only exposes 1 function [aws_sign_request(request, config)](https://github.com/awslabs/aws-crt-python/blob/191b52d307793ac105d7389b0bce646917857ed1/awscrt/auth.py#L732) for signing HTTP requests

aws-crt-java also has 1 [AwsSigningConfig](https://github.com/awslabs/aws-crt-java/blob/main/src/main/java/software/amazon/awssdk/crt/auth/signing/AwsSigningConfig.java) class, but then it has multiple functions for signing: [AwsSigner.signRequest(...)](https://github.com/awslabs/aws-crt-java/blob/dc195bbf0e698f3dd71fb1c23cdfa3db9bdea710/src/main/java/software/amazon/awssdk/crt/auth/signing/AwsSigner.java#L26), [AwsSigner.signChunk(...)](https://github.com/awslabs/aws-crt-java/blob/dc195bbf0e698f3dd71fb1c23cdfa3db9bdea710/src/main/java/software/amazon/awssdk/crt/auth/signing/AwsSigner.java#L50)

Reply:
Thanks, fixed.